### PR TITLE
Script to install ckan locally in dev and prod modes

### PR DESCRIPTION
**Pull request by aimalkhan**
implements DATA-866
* add scripts to install ckan on a local VM in both developer and production modes
* add supporting configuration files for local setup
* add supporting documentation for local setup

### CODE REVIEW

**Comment by aimalkhan** [comment]
@KatiRG updated and tested the code. This is ready for a review and test-run.

**Comment by KatiRG** [comment]
## Review Comments
Nice work! Both `ckan_localprod_install.sh` and `ckan_localprod_install.sh` work perfectly. 
I had a glitch getting Solr to install...will make a separate issue.

I have a few comments around documentation:

### README - Local dev setup
- let's pull the common tasks from the Solr and CKAN install sections into the page intro section, i.e.:
exporting SUDOPASS, installing git and cloning the repo. Maybe worth adding a note that the repo is cloned in the home directory to obtain the scripts and config files, but that it will be cloned again when running the ckan install script in the proper directory (`/usr/lib/ckan/default/src`)
- `bash postgres_install.sh` should be `sudo bash postgres_install.sh`
- for completeness, I would add the command for rebuilding the search index: `ckan -c /etc/ckan/default/ckan.ini search-index rebuild` just before the last step in the README to unset SUDOPASS (https://github.com/ongov/ckanext-ontario_theme/tree/ckan_script/scripts#running-the-script-2).

### README - Local prod setup
In the intro paragraph, I would reword it to indicate that the steps needed are taken care of by the scripts. To me it was a bit confusing as I wasn't sure if I was supposed to install all those things or if I just needed to follow the steps. Suggestion:

...This can be done by first following the steps to [install CKAN in local development mode](https://github.com/ongov/ckanext-ontario_theme/tree/ckan_script/scripts#ckanontario-theme-local-installation-in-developer-mode) to install Solr, postgreSQL and relevant databases, ckan and the ontario theme, before running `ckan_localprod_install.sh`. The production script uses a base SSL certificate from [config/nginx/local_ckan_ssl](https://github.com/ongov/ckanext-ontario_theme/blob/ckan_script/config/nginx/local_ckan_ssl), and a supervisor config file from [config/supervisor/ckan-uwsgi.conf](https://github.com/ongov/ckanext-ontario_theme/blob/ckan_script/config/supervisor/ckan-uwsgi.conf), which it copies to `/etc/nginx/sites-enabled/` and `/etc/supervisor/conf.d/ckan-uwsgi.conf`, respectively, in the local VM. The steps needed are summarized below:
...

- the hyper-link is not formatted "Install CKAN & Ontario Theme locally using instructions in [CKAN Local Install](#CKAN-&-Ontario-Theme-Local -nstallation-in-Production-Mode)."
- at the end of running `ckan_localprod_install.sh`, the user is still in the virtual env, so if they follow the README instructions and goto Step 2 `bash ckan_localprod_install.sh`, they will still be in the virtual env. Either add a `deactivate` to the end of `ckan_localprod_install.sh`, or make a note to deactivate the environment in the README steps here (e.g. at the end of Step 1 or the beginning of Step 2)
-  Also, CKAN will still be running, so add a note to ctrl-c it
- Step 2 `bash ckan_localprod_install.sh` should be `sudo bash ckan_localprod_install.sh`

Hope that is helpful. Let me know if you have any comments or feedback.

**Comment by aimalkhan** [comment]
@Alexandra-Green I fixed the script and updated the README. This is ready for review.

File: scripts/README.md [hunk 1]
```diff
@@ -1,39 +1,52 @@
...
 ```bash
-# Update packages
-echo $SUDOPASS | sudo -S -k apt-get update
+cd ckanext-ontario_theme\scripts
+```
+
+## Solr Installation
 
-# Install Git
-{ echo $SUDOPASS; echo 'Y'; } | sudo -S -k apt-get install git
```
Comment (Alexandra-Green) [inline scripts/README.md:29]: It would be best to keep this rather than the current line 20
image

File: scripts/README.md [hunk 2]
```diff
@@ -1,39 +1,52 @@
...
+git clone https://github.com/ongov/ckanext-ontario_theme.git
 ```
 
-**Step 2:** Install Git, clone the repository, fetch the `ckan_2.9_upgrade` branch, switch to the scripts directory, and execute the script. Follow the commands below:
+4. change to the scripts directory
 
 ```bash
-# Update packages
-echo $SUDOPASS | sudo -S -k apt-get update
+cd ckanext-ontario_theme\scripts
```
Comment (Alexandra-Green) [inline scripts/README.md:32]:
```suggestion
cd ckanext-ontario_theme/scripts
```

image

File: scripts/README.md [hunk 3]
```diff
@@ -91,23 +104,57 @@ api_token = "your_api_key_here"
...
+```bash
+bash setup_ckan_local.sh
 ```
 
-The script processes the CSV files and creates organizations in your CKAN application. It also uploads and attaches any resources associated with the organizations. 
+4. Rebuild SOLR search index
 
-Monitor the script output for any error messages or failed resource uploads. After completion, check if the organizations and their associated resources were created successfully in your CKAN application.
+```bash
+ ckan -c /etc/ckan/default/ckan.ini search-index rebuild
```
Comment (Alexandra-Green) [inline scripts/README.md:150]:
```suggestion
 . /usr/lib/ckan/default/bin/activate
 ckan -c /etc/ckan/default/ckan.ini search-index rebuild
```

image

File: scripts/README.md [hunk 4]
```diff
@@ -91,23 +104,57 @@ api_token = "your_api_key_here"
...
 
-1. Open a terminal and navigate to the repository directory.
-2. Execute the script with the following command:
+**Command line:**
+
+1. Do the steps common to the scripts in [common steps](#common-steps)
+2. setup solr and postgresql
+
+```bash
+sudo bash setup_solr.sh && sudo bash setup_postgres_local.sh
```
Comment (Alexandra-Green) [inline scripts/README.md:138]: running setup_solr.sh will give an error, I would suggest consolidating the solr installation above with this "running the script" section

File: scripts/README.md [hunk 5]
```diff
@@ -1,39 +1,52 @@
 # Installation Scripts Guide
 
-This document provides instructions for using bash scripts to install Ontario CKAN 2.9 and Solr 8.11.2 on individual Ubuntu 20.04 virtual machines.
+Bash scripts to install CKAN 2.9 compliant Ontario Theme, and Solr 8.11.2 on a separate Ubuntu 20.04 virtual machine. The objective of the scripts is to create a local dev-test environment.
 
-## Solr Installation
+During the course of the script execution, the code repository will be cloned into `/usr/lib/ckan/default/src`for installation. This directory and it being cloned are independent of the code used for installation purposes. This is the directory where the code changes need to be made for dev work.
```
Comment (Alexandra-Green) [inline scripts/README.md:5]:
```suggestion
During the course of the script execution, the code repository will be cloned into `/usr/lib/ckan/default/src` for installation. This directory and it being cloned are independent of the code used for installation purposes. `/usr/lib/ckan/default/src` is the directory where the code changes need to be made for dev work.
```

File: scripts/README.md [hunk 6]
```diff
@@ -91,23 +104,57 @@ api_token = "your_api_key_here"
...
+5. Unset the SUDOPASS environment variable:
 
-If any errors occur during resource uploads, review the error messages and the `failed_resource_writes` list in the script output. Address any issues related to resource files or network connectivity. 
+```bash
+unset SUDOPASS
+```
 
-You may modify the script further to meet specific requirements or change its behavior as needed.
+**Browser:**
+Check ckan status by going to `localhost` in a browser
```
Comment (Alexandra-Green) [inline scripts/README.md:160]: Can we move this section before the load organization script? The order it is currently in would be confusing to people following the installation process.

**Review by Alexandra-Green — CHANGES_REQUESTED** [review]

File: scripts/README.md [hunk 7]
```diff
@@ -1,39 +1,52 @@
...
+
+```bash
+sudo -y apt-get install git
 ```
-export SUDOPASS='your_sudo_password'
+
+3. Clone this repository
+
+```bash
+git clone https://github.com/ongov/ckanext-ontario_theme.git
```
Comment (Alexandra-Green) [inline scripts/README.md:26]: Should this be ssh? When pushing to the repository, devs have to change to ssh.

File: scripts/README.md (hunk 1 above)
Comment (aimalkhan) [reply scripts/README.md:29]: done

File: scripts/README.md (hunk 5 above)
Comment (aimalkhan) [reply scripts/README.md:5]: 👍 improved readability here.

File: scripts/README.md (hunk 7 above)
Comment (aimalkhan) [reply scripts/README.md:26]: this one, no. It is being used for installation. The one cloned by the script into `/usr/lib/ckan/default/src` is through ssh.

File: scripts/README.md (hunk 2 above)
Comment (aimalkhan) [reply scripts/README.md:32]: 👍 
(windows vs linux error...)

File: scripts/README.md (hunk 3 above)
Comment (aimalkhan) [reply scripts/README.md:150]: 👍

File: scripts/README.md (hunk 6 above)
Comment (aimalkhan) [reply scripts/README.md:160]: done.

File: scripts/README.md (hunk 4 above)
Comment (aimalkhan) [reply scripts/README.md:138]: this line now refers to "Solr Installation" section, to properly install Solr8. Thanks for pointing out.

File: scripts/README.md [hunk 8]
```diff
@@ -66,6 +79,64 @@ You can verify the installation success via command line or a browser.
...
+**Command line:**
+
+1. Do the steps common to the scripts in [common steps](#common-steps)
+2. Setup Solr8 locally as described in [Solr Installation](#solr-installation)
+
+3. Setup PostgreSQL locally
+
+```bash
+sudo bash setup_postgres_local.sh
+```
```
Comment (Alexandra-Green) [inline scripts/README.md:116]: I would either add the export sudopass line before this or remove the unset sudopass from line 55-58 (the solr step)

**Review by Alexandra-Green — APPROVED** [review]
Next steps (a different ticket), would be instructions on creating a sysadmin so you can change the layout of the homepage. This currently runs well and I'm not getting any errors.
